overlord/servicestate/service_control_test.go

```diff
@@ -476,3 +476,71 @@ func (s *serviceControlSuite) TestConflict(c *C) {
 	_, err := snapstate.Remove(st, "test-snap", snap.Revision{}, nil)
 	c.Assert(err, ErrorMatches, `snap "test-snap" has "service-control" change in progress`)
 }
+
+func (s *serviceControlSuite) TestUpdateSnapstateServices(c *C) {
+	var tests = []struct {
+		enable                    []string
+		disable                   []string
+		expectedSnapstateEnabled  []string
+		expectedSnapstateDisabled []string
+		changed                   bool
+	}{
+		{
```

Comment:
maybe also add a test case for enable and disable together returning an error?

Reply:
This is tested below in line 545.